### PR TITLE
[RFC] [3.7] Add basic custom fields support

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Add basic custom fields support

### Testing Instructions

Apply this patch on a 3.7 version

### Expected result

Confirm you can add an fill the custom fields, there is no rendering done in the frontend yet

### Actual result

No custom fields support

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by astridx** [comment]
I tested this succeful a few days ago. It was possible to add and edit custom fields in the backend. 
But this fields are not shown in the frontend. Should we merge this stand and add the frondend support in an separat pr?

**Comment by hakanara** [comment]
I have some menu links like: index.php?option=com_weblinks&view=category&id=71
I would like to use fields in these menu items. I think there is no progress so far in the addition of weblinks. Am I right? Are you planing to do so?

**Comment by yvesh** [comment]
@hakanara yes it's planned for Weblinks 3.7 - the thing is we are missing testers here and there is a merge conflict in this patch.

@astridx can you remove the merge conflict.. Going to test it than in the next days

**Comment by zero-24** [comment]
> @zero-24 can you remove the merge conflict? Going to test it than in the next days

done.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on 8b9ab398a9e9eeb72bd1f2ac6f117f050ebf25dc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/317">issues.joomla.org/tracker/weblinks/317</a>.</sub>

File: src/administrator/components/com_weblinks/config.xml [hunk 1]
```diff
@@ -411,6 +411,21 @@
 			<option value="1">JSHOW</option>
 			<option value="0">JHIDE</option>
 		</field>
+<<<<<<< HEAD
```
Comment (astridx) [inline src/administrator/components/com_weblinks/config.xml]: Here is a merge conflict hint not deleted

**Comment by hakanara** [comment]
Is there anything that I can do now, like testing something?

**Comment by astridx** [comment]
@hakanara That would be great if you would test.

**Comment by franz-wohlkoenig** [comment]
@astridx will test tomorrow.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on e69347e2558f3887ad177ee8a60af304592e6757<br /><br />I can add and fill Custom Fields, no Rendering in the Frontend.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/317">issues.joomla.org/tracker/weblinks/317</a>.</sub>

**Comment by astridx** [comment]
@franz-wohlkoenig Thank you.
@all
Does anyone have anything against it when I merge that?

**Comment by franz-wohlkoenig** [comment]
@astridx is a Frontend-Rendering coming?

**Comment by astridx** [comment]
@franz-wohlkoenig 
Yes frontend rendering will work, when we have the single view from this pr https://github.com/joomla-extensions/weblinks/pull/349

**Comment by hakanara** [comment]
Thank you very much all. I will be able to test it soon. Before that.can I ask you if the frontend rendering will be available only for the single item? I need it for weblinks category view as well. 

Kind regards.

**Comment by hakanara** [comment]
I wanted to test this, I downloaded and installed Joomla 3.8.1-rc2-dev. I have downloaded and installed Joomla patch tester. Then I have retrieved the 4 pull requests. But when I apply patch for this item, I get this error:
The file marked for modification does not exist: src/administrator/components/com_weblinks/access.xml

Why is this and what can I do test this patch?

And also, I have asked a question above which remained unanswered. Will the fields functionality be working for new weblinks for the links like: index.php?option=com_weblinks&view=category&id=71
I need frontend fields rendering for the weblinks category page as well.

Kind regards.

**Comment by astridx** [comment]
Hi @hakanara 
thank you very much.


> The file marked for modification does not exist: src/administrator/components/com_weblinks/access.xml
> Why is this and what can I do test this patch?

You also have to install weblinks as it is not integrated in Joomla! any more.



> Will the fields functionality be working for new weblinks for the links like: index.php?option=com_weblinks&view=category&id=71
> I need frontend fields rendering for the weblinks category page as well.

I'm not 100% sure, but I mean yes. A test will show you. 

Note, the single view is not part of this PR. You find this in this PR: https://github.com/joomla-extensions/weblinks/pull/349

**Comment by hakanara** [comment]
Hi @astridx I have also installed weblinks pkg-weblinks-3.6.0 from Joomla offical extensions on top of the installation that I have stated above but the issue still persists with the same error.

**Comment by franz-wohlkoenig** [comment]
I guess you have to install pkg-weblinks-3.7.0-beta1.zip

**Comment by hakanara** [comment]
Danke @franz-wohlkoenig I also installed that and I still get the same error. Very strange. Is there any other way to upload the patch and see if it works or not?

**Comment by astridx** [comment]
> Is there any other way to upload the patch and see if it works or not?

You can apply the patch via git as explained here:
https://github.com/joomla-extensions/weblinks
But then you have to build the installation package from the data in the src directory or use the tests to create this.

Do you see a  file with the name access.xml in your folder JOOMLA/administrator/components/com_weblinks/access.xml ?

**Comment by hakanara** [comment]
The access.xml file exists in that directory:
JOOMLA/administrator/components/com_weblinks/access.xml

But theere is no folder called JOOMLA/**src**/administrator/components/com_weblinks/access.xml

I will see if I can create an installation package from those files there. Thank you very much.

**Comment by astridx** [comment]
> But theere is no folder called JOOMLA/src/administrator/components/com_weblinks/access.xml

I think this is the file that patch tester want to take out of the weblinks repo for overriding your joomla weblink installation.

**Comment by hakanara** [comment]
OK, I have downloaded the weblinks-master.zip file, extracted that, copied all the src directory to the root of my Joomla directory. And finally I managed to apply the patch/es. 

I don't want to mislead anyone but after the aaplication, I didn't see any "fields" menu link under weblinks menü in administration. Also I clicked on "new" to see if there will be any fields tab when I create a new single web link or a web link category.

If you need further assistance, I will be available for testing the issue. Thank you very much for all the contributers for your time and efforts

**Comment by hakanara** [comment]
Hello. Is there any progress for adding the custom fields for weblinks?

**Comment by astridx** [comment]
Yes, this PR is merged. But we do not have a release up to now.
If you like you can install web links **for testing** with the zip file from this comment: https://github.com/joomla-extensions/weblinks/pull/377#issuecomment-342606904
If there is more interest, maybe there will be a new release sooner.

**Comment by franz-wohlkoenig** [comment]
I'm interested.

**Pull request merged**